src/VisualStudio/Core/Test/ObjectBrowser/CSharp/ObjectBrowerTests.vb

```diff
@@ -1196,6 +1196,111 @@ ServicesVSResources.Returns_colon & vbCrLf &
             End Using
         End Sub
 
+        <ConditionalFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.ObjectBrowser)>
+        Public Sub TestDescription_XmlDocComment_Returns1()
+            Dim code =
+<Code>
+    <![CDATA[
+class C
+{
+    /// <summary>
+    /// Describes the method.
+    /// </summary>
+    /// <returns>Returns a value.</returns>
+    public int M()
+    {
+        return 0;
+    }
+}
+]]>
+</Code>
+
+            Using state = CreateLibraryManager(GetWorkspaceDefinition(code))
+                Dim library = state.GetLibrary()
+                Dim list = library.GetProjectList()
+                list = list.GetTypeList(0)
+                list = list.GetMemberList(0)
+
+                list.VerifyImmediateMemberDescriptions(
+"public int M()" & vbCrLf &
+$"    {String.Format(ServicesVSResources.Member_of_0, "C")}" & vbCrLf &
+"" & vbCrLf &
+ServicesVSResources.Summary_colon & vbCrLf &
+"Describes the method." & vbCrLf &
+"" & vbCrLf &
+ServicesVSResources.Returns_colon & vbCrLf &
+"Returns a value.")
+            End Using
+        End Sub
+
+        <ConditionalFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.ObjectBrowser)>
+        Public Sub TestDescription_XmlDocComment_Returns2()
+            Dim code =
+<Code>
+    <![CDATA[
+class C
+{
+    /// <summary>
+    /// Gets a value.
+    /// </summary>
+    /// <returns>Returns a value.</returns>
+    public int M { get; }
+}
+]]>
+</Code>
+
+            Using state = CreateLibraryManager(GetWorkspaceDefinition(code))
+                Dim library = state.GetLibrary()
+                Dim list = library.GetProjectList()
+                list = list.GetTypeList(0)
+                list = list.GetMemberList(0)
+
+                list.VerifyImmediateMemberDescriptions(
+"public int M { get; }" & vbCrLf &
+$"    {String.Format(ServicesVSResources.Member_of_0, "C")}" & vbCrLf &
```

Comment:
Should we have a test helper for building the description string so that the formatting logic is in a single place instead of in each test? something like
```csharp
string BuildDescription(string declaration, string containingMemberName = null, string summaryText = null, string returnText = null, string valueText = null)
```

Reply:
We could, but it seems out of scope for this pull request